osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -132,11 +141,39 @@ private void updateStatistics()
                     break;
             }
 
-            starDifficulty.Value = ((float)(Beatmap?.StarDifficulty ?? 0), null);
-
             HpDrain.Value = (baseDifficulty?.DrainRate ?? 0, adjustedDifficulty?.DrainRate);
             Accuracy.Value = (baseDifficulty?.OverallDifficulty ?? 0, adjustedDifficulty?.OverallDifficulty);
             ApproachRate.Value = (baseDifficulty?.ApproachRate ?? 0, adjustedDifficulty?.ApproachRate);
+
+            updateStarDifficulty();
+        }
+
+        private IBindable<StarDifficulty> normalStarDifficulty;
+        private IBindable<StarDifficulty> moddedStarDifficulty;
+        private CancellationTokenSource starDifficultyCancellationSource;
+
+        private void updateStarDifficulty()
+        {
+            starDifficultyCancellationSource?.Cancel();
+
+            if (Beatmap == null)
+                return;
+
+            var ourSource = starDifficultyCancellationSource = new CancellationTokenSource();
```

Comment:
Reason for `ourSource` local? Doesn't seem like it could be thread-unsafe to just use the field.

Reply:
Old code, whoops.